flaml/model.py

```diff
@@ -578,23 +567,47 @@ def _compute_metrics_by_dataset_name(self, eval_pred):
                 )
             }
         else:
-            from .nlp.utils import load_default_huggingface_metric_for_task
-            import datasets
-
-            default_metric_name = load_default_huggingface_metric_for_task(self._task)
+            (
+                default_metric_name,
+                default_metric_mode,
+            ) = load_default_huggingface_metric_for_task(self._task)
             metric = datasets.load_metric(default_metric_name)
-            return {
-                "val_loss": metric.compute(predictions=predictions, references=labels)[
-                    default_metric_name
-                ]
-            }
+            if default_metric_mode == "max":
+                return {
+                    "val_loss": -metric.compute(
+                        predictions=predictions, references=labels
+                    )[default_metric_name]
+                }
+            else:
+                return {
+                    "val_loss": metric.compute(
+                        predictions=predictions, references=labels
+                    )[default_metric_name]
+                }
 
-    def predict(self, X_test):
-        from datasets import Dataset
-        from .nlp.utils import load_model
-        from transformers import TrainingArguments
-        from .nlp.huggingface.trainer import TrainerForAuto
+    def predict_proba(self, X_test):
+        assert (
+            self._task in CLASSIFICATION
+        ), "predict_proba is only available in classification tasks"
+        if X_test.dtypes[0] == "string":
+            X_test = self._preprocess(X_test, self._task, **self._kwargs)
+            test_dataset = Dataset.from_pandas(X_test)

```

Comment:
This doesn't look right.
line 592 and 594 should be inside _preprocess().
Make sure test_dataset is defined in all cases as it is needed in line 607.